src/OAuth/Authorizer.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace ZfrShopify\OAuth;
+
+use GuzzleHttp\Client as HttpClient;
+use Psr\Http\Message\ResponseInterface;
+use Zend\Diactoros\Response\RedirectResponse;
+use ZfrShopify\Exception;
+
+/**
+ * Simplify the OAuth dance of Shopify
+ *
+ * Especially, this class can be used to create an authorization response, and exchanging a temporary code
+ * against a long-lived access token
+ *
+ * @author Michaël Gallego
+ */
+class Authorizer
+{
+    /**
+     * Create an authorization redirection request for a given app
+     *
+     * Please note that this method will automatically generate a nonce value. You are responsible to
+     * persist it in database, and validate it during the OAuth dance
+     *
+     * @param  string $apiKey
+     * @param  string $shopDomain
+     * @param  array  $scopes
+     * @param  string $redirectionUri
+     * @param  string $nonce
+     * @return ResponseInterface
+     * @throws Exception\MissingApiKeyException
+     */
+    public function createAuthorizationResponse(string $apiKey, string $shopDomain, array $scopes, string $redirectionUri, string $nonce)
```

Comment:
And how should I call the class that do the exchange? Those two operations are really linked together :)